### ISSUE TITLE
Values always zero while training on GPU

### ISSUE THREAD

**Issue by madingo87** [Microsoft/CNTK#615]
I have a Problem with training networks on GPU.

When i run (for example) the 3Classes_bs.cntk from the tutorial the evaluation and criterion-values are always 0. Training with CPU works just fine. It says in the log that GPU 0 is locked for use.

> CNTKCommandTrainBegin:
> 
>  Train
> LockDevice: Locked GPU 0 to test availability.
> LockDevice: Unlocked GPU 0 after testing.
> LockDevice: Locked GPU 0 for exclusive use.
> 
> Creating virgin network.
> 
> Post-processing network...
> 
> 3 roots:
>         ce = CrossEntropyWithSoftmax()
>         errs = ErrorPrediction()
>         z = Plus()

and here is a snippet from the log while training

> Starting Epoch 4: learning rate per sample = 0.040000  effective momentum = 0.900000  momentum as time constant = 237.3 samples
> BlockRandomizer::StartEpoch: epoch 3: frames [3000..4000](first sequence at sample 3000), data subset 0 of 1
> 
> Starting minibatch loop.
> Finished Epoch[ 4 of 50]: [Training] ce = 0.00000000 \* 1000; errs = 0.000% \* 1000; totalSamplesSeen = 4000; learningRatePerSample = 0.039999999; epochTime=0.033761s
> SGD: Saving checkpoint model 'Models/MC.dnn.4'
> 
> Starting Epoch 5: learning rate per sample = 0.040000  effective momentum = 0.900000  momentum as time constant = 237.3 samples
> BlockRandomizer::StartEpoch: epoch 4: frames [4000..5000](first sequence at sample 4000), data subset 0 of 1
> 
> Starting minibatch loop.
> Finished Epoch[ 5 of 50]: [Training] ce = 0.00000000 \* 1000; errs = 0.000% \* 1000; totalSamplesSeen = 5000; learningRatePerSample = 0.039999999; epochTime=0.034928s
> SGD: Saving checkpoint model 'Models/MC.dnn.5'
> 
> Starting Epoch 6: learning rate per sample = 0.040000  effective momentum = 0.900000  momentum as time constant = 237.3 samples
> BlockRandomizer::StartEpoch: epoch 5: frames [5000..6000](first sequence at sample 5000), data subset 0 of 1
> 
> Starting minibatch loop.
> Finished Epoch[ 6 of 50]: [Training] ce = 0.00000000 \* 1000; errs = 0.000% \* 1000; totalSamplesSeen = 6000; learningRatePerSample = 0.039999999; epochTime=0.034674s
> SGD: Saving checkpoint model 'Models/MC.dnn.6'

what am i doing wrong? what do i have to check or do i miss?
Thanks for any hint!

**Comment by jotarun** [comment]
most likely unsupported GPU. See closed question #591

**Comment by madingo87** [comment]
Thanks a lot! I couln't find this one... 
I will try to confirm that as soon as possible!

**Comment by frankseide** [comment]
Yes. The message "Locked GPU 0" means that you are running on the GPU 0, which has been exclusively locked for this CNTK process.

**Comment by madingo87** [comment]
Okay, i found out, that my GPU has the Compute Capability 2.1 
I thought that could be the problem, but i was indeed confused by the log saying "Locked GPU 0 for exclusive use". (I thought maybe thats the first index of the GPUs)

Thanks for the awesome support here! Highly appreciate it!
